importer/helper.go

```diff
@@ -257,3 +245,44 @@ func (paths *blockTarPaths) Swap(i, j int) {
 	(*paths)[i] = (*paths)[j]
 	(*paths)[j] = t
 }
+
+// GetGenesisFile creates a reader from given the genesis file
+func GetGenesisFile(genesisJSONPath string, client *algod.Client, l *log.Logger) io.Reader {
+	var genesisReader io.Reader
+	var err error
+	if genesisJSONPath != "" {
+		// Read file if specified.
+		l.Infof("loading genesis file %s", genesisJSONPath)
+		genesisReader, err = os.Open(genesisJSONPath)
+		maybeFail(err, l, "unable to read genesis file %s", genesisJSONPath)
+	} else if client != nil {
+		// Fallback to asking algod for genesis if file is not specified.
+		l.Infof("fetching genesis from algod")
+		genesisString, err := client.GetGenesis().Do(context.Background())
+		maybeFail(err, l, "unable to fetch genesis from algod")
+		genesisReader = strings.NewReader(genesisString)
+	} else {
+		l.Fatal("Neither genesis file path or algod client provided for initial import.")
+	}
+	return genesisReader
+}
+
+func checkGenesisHash(db idb.IndexerDb, genesisReader io.Reader) error {
+	network, err := db.GetNetworkState()
+	if err != nil {
+		return fmt.Errorf("unable to fetch network state from db %v", err)
```

Comment:
it's better to use `%w` instead of `%v`, so that `err` can be extracted later on